src/Gaufrette/Adapter/ChecksumCalculator.php

```diff
@@ -10,11 +10,12 @@
 interface ChecksumCalculator
 {
     /**
-     * Returns the checksum of the specified key.
+     * Returns the checksum of the specified key and algorithm.
      *
      * @param string $key
+     * @param string $algo
      *
      * @return string
      */
-    public function checksum($key);
+    public function checksum($key, $algo);
```

Comment:
* Unfortunately, excluding `Local` adapter, there are many other places where the change needs to be done
* It would be a BC break for anyone having an adapter outside of this project
* It brings a new dependency, `hash` ext

I'm not against the idea. But I don't think we should do this right now, on `master`. It might better suite the refacto prepared on `split-adapters` branch.